### PR TITLE
sub menu improvements

### PR DESCRIPTION
**Pull request by TobyMosque**

### CODE REVIEW

File: src/components/Menu.vue [hunk 1]
```diff
@@ -34,9 +34,16 @@ export default defineComponent({
 
     onMounted(async () => {
       try {
-        const response = await api.get(
-          '/menus?populate[0]=children&populate[children][populate][0]=children&populate[children][populate][children][populate][0]=children&populate[children][populate][children][populate][children][populate][0]=children&populate[children][populate][children][populate][children][populate][children][populate][0]=children&populate[children][populate][children][populate][children][populate][children][populate][children][populate][0]=children&populate[children][populate][children][populate][children][populate][children][populate][children][populate][children][populate][0]=children&pagination[pageSize]=100'
-        );
+        const query = new URLSearchParams({
```
Comment (TobyMosque) [inline src/components/Menu.vue:37]: A suggestion: avoid using very long query strings. An alternative is to use URLSearchParams, which is useful both for making them more readable and for constructing them dynamically.

File: src/components/SubMenu.vue [hunk 2]
```diff
@@ -1,17 +1,25 @@
 <template>
-  <q-menu auto-close>
+  <q-menu ref="qMenu" :anchor="anchor" :self="self">
```
Comment (TobyMosque) [inline src/components/SubMenu.vue:2]: You may want to position the main menu below the toolbar and the submenus next to the main menu. To achieve this, we need to propagate the anchor and self.

File: src/components/SubMenu.vue [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
     <q-list style="min-width: 100px">
       <q-item
         clickable
         v-for="menu in menuItem.attributes.children.data"
         :key="menu.id"
+        @click="close(menu)"
```
Comment (TobyMosque) [inline src/components/SubMenu.vue:8]: `auto-close` is useful only when ALL menu options should close the menu, which is not the current case. Therefore, we need to control who and when can close the menu.

File: src/components/SubMenu.vue [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
+        <template v-if="menu.attributes.children.data.length > 0">
+          <q-item-section side>
+            <q-icon name="keyboard_arrow_right" />
+          </q-item-section>
+          <SubMenu
+            :menuItem="menu"
+            :level="level + 1"
+            anchor="top end"
+            self="top start"
+            @close="close"
```
Comment (TobyMosque) [inline src/components/SubMenu.vue:20]: When closing a submenu, we should propagate the close event to all menus in the hierarchy.

File: src/components/SubMenu.vue [hunk 5]
```diff
@@ -28,23 +37,36 @@ export default defineComponent({
       type: Object as PropType<MenuItem>,
       required: true,
     },
+    anchor: Object as PropType<QMenu['anchor']>,
```
Comment (TobyMosque) [inline src/components/SubMenu.vue:40]: `anchor` and `self` would be of the same type as the `QMenu` properties with the same names.

File: src/components/SubMenu.vue [hunk 6]
```diff
@@ -28,23 +37,36 @@ export default defineComponent({
...
+  setup(props, { emit }) {
+    const qMenu = ref<QMenu>();
     const locale = ref<string>('en');
 
     function getLocalizedTitle(menu: MenuItem): string {
       const titleKey = `title_${locale.value}`;
       return menu.attributes[titleKey as keyof MenuItemAttributes] as string;
     }
 
+    function close(menu?: MenuItem) {
```
Comment (TobyMosque) [inline src/components/SubMenu.vue:54]: Here we are checking if the menu has submenus; if it does, abort the close operation.

File: src/components/Menu.vue (hunk 1 above)
Comment (gluneau) [reply src/components/Menu.vue:37]: Very cool, yes this is smart and makes it much simpler

File: src/components/SubMenu.vue (hunk 5 above)
Comment (gluneau) [reply src/components/SubMenu.vue:40]: That's the magic, I did not see that coming at all.  Great job

**Review by gluneau — APPROVED** [review]

File: src/components/SubMenu.vue (hunk 6 above)
Comment (gluneau) [reply src/components/SubMenu.vue:54]: Amazing. Big Thanks.

**Pull request merged**